klient/conf/config.go

```diff
@@ -0,0 +1,124 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conf
+
+import (
+	"flag"
+	"os"
+	"os/user"
+	"path"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var DefaultClusterContext = ""
+
+// New returns Kubernetes configuration value of type *rest.Config.
+func New(fileName string) (*rest.Config, error) {
+	// if filename is not provided assume in-cluster-config
+	if fileName == "" {
+		return rest.InClusterConfig()
+	}
+
+	// create the config object from k8s config path
+	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: fileName}, &clientcmd.ConfigOverrides{}).ClientConfig()
+}
+
+// NewWithContextName returns k8s config value of type *rest.Config
+func NewWithContextName(fileName, context string) (*rest.Config, error) {
+	// create the config object from k8s config path and context
+	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: fileName},
+		&clientcmd.ConfigOverrides{
+			CurrentContext: context,
+		}).ClientConfig()
+}
+
+// ResolveKubeConfigFile returns the kubeconfig file from
+// either flag --kubeconfig or env KUBECONFIG.
+// It will only call flag.Parse() if flag.Parsed() is false.
+// If --kubeconfig, or KUBECONFIG, or  $HOME/.kube/config not provided then
+// assume in cluster.
+func ResolveKubeConfigFile() string {
+	var kubeConfigPath string
+
+	// If a flag --kubeconfig  is specified with the config location, use that
+	if flag.Parsed() {
+		f := flag.Lookup("kubeconfig")
+		if f != nil {
+			return f.Value.String()
+		}
+	}
+
+	// if KUBECONFIG env is defined then use that
+	kubeConfigPath = os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
+	if kubeConfigPath != "" {
+		return kubeConfigPath
+	}
+
+	var (
+		homeDir string
+		ok      bool
+	)
+
+	// check if $HOME/.kube/config is present
+	// if $HOME is unset, get it from current user
+	if homeDir, ok = os.LookupEnv("HOME"); !ok {
+		u, err := user.Current()
+		if err != nil {
+			// consider it as in-cluster-config
+			return ""
+		}
+
+		kubeConfigPath = path.Join(u.HomeDir, clientcmd.RecommendedHomeDir, clientcmd.RecommendedFileName)
+	} else {
+		kubeConfigPath = path.Join(homeDir, clientcmd.RecommendedHomeDir, clientcmd.RecommendedFileName)
+	}
+
+	// check if the config path exists
+	if exists(kubeConfigPath) {
+		return kubeConfigPath
+	}
+
+	return ""
+}
+
+// ResolveClusterContext returns cluster context name based on --context flag.
+func ResolveClusterContext() string {
+	// If a flag --context is specified use that
+	if flag.Parsed() {
+		f := flag.Lookup("context")
+		if f != nil {
+			return f.Value.String()
+		}
+	}
+
+	return ""
+}
+
+// exists reports whether the named file or directory exists.
+func exists(name string) bool {
```

Comment:
Wondering if `path` or `filePath` could be a better name for parameter. Also helps if specify if this expects absolute paths or relative paths